lib/rust/ensogl/component/slider/src/lib.rs

```diff
@@ -0,0 +1,416 @@
+                |(value, min, max)| value.clamp(*min, *max)
+            );
+            output.value            <+ value;
+
+            track_pos.target        <+ all3(&value, &input.set_value_min, &input.set_value_max).map(
+                |(value, min, max)| (value - min) / (max - min)
+            );
+
+
+            // model update
+
+            eval track_pos.value (
+                (v) model.track.value.set(*v);
+            );
+            component_size <- all2(&input.set_width, &input.set_height).map(
+                |(width, height)| Vector2::new(*width, *height)
+            );
+            eval component_size (
+                (size) model.set_size(*size);
+            );
+
+            value_is_default        <- all2(&value, &input.set_value_default).map(
+                |(value, default)| value==default
+            );
+            value_is_default_true   <- value_is_default.on_true();
+            value_is_default_false  <- value_is_default.on_false();
+            eval_ value_is_default_true ({
+                model.value_left.set_property_default(formatting::Weight::Normal);
+                model.value_dot.set_property_default(formatting::Weight::Normal);
+                model.value_right.set_property_default(formatting::Weight::Normal);
+            });
+            eval_ value_is_default_false ({
+                model.value_left.set_property_default(formatting::Weight::Bold);
+                model.value_dot.set_property_default(formatting::Weight::Bold);
+                model.value_right.set_property_default(formatting::Weight::Bold);
+            });
+
+
+            // colors
+
+            background_color.target <+ all2(&input.set_background_color, &input.set_slider_enabled).map(
+                |(color, enabled)| if *enabled { *color } else { color.to_grayscale() }
+            );
+            track_color.target <+ all2(&input.set_slider_color, &input.set_slider_enabled).map(
+                |(color, enabled)| if *enabled { *color } else { color.to_grayscale() }
+            );
+            value_color.target <+ all2(&input.set_value_color, &input.set_slider_enabled).map(
+                |(color, enabled)| if *enabled { *color } else { color.to_grayscale() }
+            );
+            label_color.target <+ all2(&input.set_label_color, &input.set_slider_enabled).map(
+                |(color, enabled)| if *enabled { *color } else { color.to_grayscale() }
+            );
+
+            eval background_color.value (
+                (color) model.set_background_color(*color);
+            );
+            eval track_color.value (
+                (color) model.set_track_color(*color);
+            );
+            eval value_color.value (
+                (color) {
+                    model.value_left.set_property_default(color);
+                    model.value_dot.set_property_default(color);
+                    model.value_right.set_property_default(color);
```

Comment:
repeated code. I've seen already 2 other blocks of 3 lines with similar expressions.